src/lib.rs

```diff
@@ -593,6 +630,48 @@ impl<S: SuffixScheme> Write for FileRotate<S> {
                     file.write_all(buf)?;
                 }
             }
+            ContentLimit::Time(time) => {
+                let local = Local::now();
+                if let Some(modified) = self.modified {
+                    match time {
+                        TimeFrequency::Hourly => {
+                            if local - Duration::hours(1) >= modified {
+                                self.rotate()?;
+                            }
+                        }
+                        TimeFrequency::Daily => {
+                            if local.date() > modified.date() {
+                                self.rotate()?;
+                            }
+                        }
+                        TimeFrequency::Weekly => {
+                            if local - Duration::days(7) > modified {
+                                self.rotate()?;
+                            }
+                        }
+                        TimeFrequency::Monthly => {
+                            if local.format("%Y%m").to_string().parse::<i8>().unwrap_or(0)
+                                > modified
+                                    .format("%Y%m")
+                                    .to_string()
+                                    .parse::<i8>()
+                                    .unwrap_or(0)
```

Comment:
Why do we use %Y%m here? Afaict this gives `202204`, which is not a number that fits in `i8`.

Reply:
Ah, you are right, i8 is to small. I had no better idea to implement the right month. Because using only months will not work. When you change from December to January, you need the year info to.